### ISSUE TITLE
Improve error message for removing components of targets other than the host's

### ISSUE THREAD

**Issue by andradei** [rust-lang/rustup#1091]
If I `rustup target add x86_64-unknown-linux-gnu` on a _x86_64-apple-darwin_ host, a component is also installed, namely: _rust-std-x86_64-unknown-linux-gnu_.

The usability issue arrives when you try to `rustup component remove rust-std-x86_64-unknown-linux-gnu`. Which will throw the following error:

```
error: toolchain 'stable-x86_64-apple-darwin' does not contain component 'rust-std-x86_64-unknown-linux-gnu' for target 'x86_64-apple-darwin'
```

Removing that component is probably impossible (from all of the methods I tried) and the only thing that works is removing the target: `rustup target remove x86_64-unknown-linux-gnu` which also removes its components.

To improve usability, a message could accompany the error above:

"You are on a x86_64-apple-darwin system. Try `rustup target add x86_64-unknown-linux-gnu` instead."

Or something along those lines.

Thank you for `rustup`. It is an amazing tool.

**Comment by setharnold** [comment]
Perhaps if components cannot be removed separately then the `rustup component remove` subcommand should be removed?

Thanks

**Comment by andradei** [comment]
There are components that can be added and removed like target's docs (which are installed by default now), and `rust-src`.

So maybe the components that can't be removed should be signified as such instead of removing `rustup component remove`.

In other words, if there are components that can be installed via `rustup component add`, then its `remove` counterpart shouldn't be removed.

**Comment by rami3l** [comment]
I believe this is a duplicate of #3320, which is itself a duplicate of #3166, fixed in #3601:


```console
> rustup --version
rustup 1.27.1+198 (81a71647f 2024-06-09)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.78.0 (9b00956e5 2024-04-29)`
> rustup component add rust-std-aarch64-apple-darwin
info: component 'rust-std' for target 'aarch64-apple-darwin' is up to date
> rustup component add rust-std-x86_64-apple-darwin
info: downloading component 'rust-std' for 'x86_64-apple-darwin'
info: installing component 'rust-std' for 'x86_64-apple-darwin'
 23.4 MiB /  23.4 MiB (100 %)  20.7 MiB/s in  1s
> rustup component remove rust-std-x86_64-apple-darwin
info: removing component 'rust-std' for 'x86_64-apple-darwin'
```
